pkgs/development/libraries/lensfun/default.nix

```diff
@@ -21,17 +21,23 @@ stdenv.mkDerivation {
     owner = "lensfun";
     repo = "lensfun";
     rev = "v${version}";
-    sha256 = "0isli0arns8bmxqpbr1jnbnqh5wvspixdi51adm671f9ngng7x5r";
+    sha256 = "0ixf0f7qv0mc7zrw9w1sb60w833g4rqrfj8cjxwzv2vimqcksccz";
   };
 
   # replace database with a more recent snapshot
+  # the mastr branch uses version 2 profiles, while 0.3.3 requires version 1 profiles,
+  # so we run the conversion tool the project provides,
+  # then untar the verson 1 profiles into the source dir before we build
   prePatch = ''
-    rm -R ./data/db
-    cp -R ${lensfunDatabase}/data/db ./data
+    rm -R data/db
+    python3 ${lensfunDatabase}/tools/lensfun_convert_db_v2_to_v1.py $TMPDIR ${lensfunDatabase}/data/db
+    mkdir -p data/db
+    tar xvf $TMPDIR/db/version_1.tar -C data/db
+    date +%s > data/db/timestamp.txt
   '';
 
   nativeBuildInputs = [ cmake pkg-config ];
-  buildInputs = [ glib zlib libpng ];
+  buildInputs = [ glib zlib libpng python3 python3.pkgs.lxml ];
```

Comment:
This should go to `nativeBuildInputs` since it is executed at build time. Also please add it to a separate line and add a comment that it is for the db conversion.

Reply:
Done, thanks.